content/blog/deploying-the-infrastructure-of-oauth-server-for-cms-app/index.md

```diff
@@ -0,0 +1,363 @@
+                },
+                {
+                    name: "GITHUB_SECRET",
+                    value: cmsStackConfig.githubSecret
+                },
+                {
+                    name: "TARGET_PORT",
+                    value: pulumi.interpolate `${inputTargetGroupPort}`
+                },
+                {
+                    name: "GITHUB_SCOPE",
+                    value: inputGithubScope
+                },
+            ]
+        },
+    },
+    desiredCount: 1,
+});
+```
+
+For `HOST` we would pass in the link that we would want this OAuth Server to have. We could set it by set the stack configuration by doing `pulumi config set pulumi-website-cms:targetDomain https://some-cms-oauth-domain.pulumi-demos.com` and replace the name with what you want the OAuth Server to have.
+
+`SESSION_SECRET` is another environment variable to pass in main.go that we mention earlier. We generated value by using Pulumi's `random.RandomPassword` previously.
+
+To get `GITHUB_SECRET` and `GITHUB_TOKEN`, we need to register for a Github OAuth Application. Netlify have provided [instructions](https://docs.netlify.com/visitor-access/oauth-provider-tokens/#setup-and-settings) for that.
+For the Home Page Url should be the targetDomain that we build with our [previous post]({{< relref "/blog/deploying-netlify-cms-on-aws" >}}).
+For the Authorization callback URL we should enter `https://{{the domain of your OAuth App}}/callback/github` which is specified by the main.go.
+
+```go
+    github.New(
+        os.Getenv("GITHUB_KEY"), os.Getenv("GITHUB_SECRET"),
+        // concatenate with the host name
+        fmt.Sprintf("%s/callback/github", host),
+        "public_repo",
+    )
+```
+
+If you are using different storaging system like Bitbucket and Gitlab, you can specify callback url to be `https://{{the domain of your OAuth App}}/callback/{{ backend name }}` and relace `{{backend name}}` with `bitbucket` or `github` correspondingly.
+
+Then we can copy and paste the github key and secret as stack configuration as well.
+
+```bash
+$ pulumi config set netlify-cms-oauth-provider-infrastructure:githubKey {{YOUR_GITHUB_KEY}}
+$ pulumi config set --secret netlify-cms-oauth-provider-infrastructure:githubSecret
+$ {{YOUR_GITHUB_SECRET}}
+```
+
+`--secret` would safely encrypt your github secret. Also,
+don't directly append the secret to the command like this `$ pulumi config set --secret netlify-cms-oauth-provider-infrastructure:githubKey {{YOUR_GITHUB_SECRET}}`.
+Because it might cause the secret to being stored inside the command memory. We should only specify the key without a name and hit ENTER key, then you can type secret on next line(won't show the value)
+
+For `TARGET_PORT` that is how we passed in the `targetGroupPort` configuration value to main.go. That is also the port number that we want the server itself to serve. We also checked if this environment variable is set in main.go in case we need to test main.go locally without Pulumi.
```

Comment:
```suggestion
We pass `TARGET_PORT` to `targetGroupPort` which is the port used to serve the application. We also check if this variable is set in main.go in case we need to test locally without Pulumi.
```